src/main/java/gregtech/api/unification/material/registry/IMaterialRegistryManager.java

```diff
@@ -0,0 +1,96 @@
+package gregtech.api.unification.material.registry;
+
+import gregtech.api.GTValues;
+import gregtech.api.unification.material.Material;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+
+public interface IMaterialRegistryManager {
+
+    /**
+     * Create a registry for a modid. Accessible when in phase {@link Phase#PRE}.
+     *
+     * @param modid the mod id for the registry
+     * @return the registry for the mod
+     */
+    @Nonnull
+    MaterialRegistry createRegistry(@Nonnull String modid);
+
+    /**
+     * Get a mod's registry. Accessible during all phases.
+     *
+     * @param modid the modid of the mod
+     * @return the registry associated with the mod, or the GregTech registry if it does not have one
+     */
+    @Nonnull
+    MaterialRegistry getRegistry(@Nonnull String modid);
+
+    /**
+     * Get a mod's registry. Accessible during all phases.
+     *
+     * @param networkId the network ID of the registry
+     * @return the registry associated with the network ID, or the GregTech registry if it does not have one
+     */
+    @Nonnull
+    MaterialRegistry getRegistry(int networkId);
+
+    /**
+     * Accessible when in phases:
+     * <ul>
+     * <li>{@link Phase#OPEN}</li>
+     * <li>{@link Phase#CLOSED}</li>
+     * <li>{@link Phase#FROZEN}</li>
+     * </ul>
+     *
+     * @return all the Material Registries
+     */
+    @Nonnull
+    Collection<MaterialRegistry> getRegistries();
+
+    /**
+     * Accessible when in phases:
+     * <ul>
+     * <li>{@link Phase#CLOSED}</li>
+     * <li>{@link Phase#FROZEN}</li>
+     * </ul>
+     *
+     * @return all registered materials.
+     */
+    @Nonnull
+    Collection<Material> getRegisteredMaterials();
+
+    /**
+     * Get a material from a String in formats:
+     * <ul>
+     * <li>{@code "modid:registry_name"}</li>
+     * <li>{@code "registry_name"} - where modid is inferred to be {@link GTValues#MODID}</li>
+     * </ul>
+     *
+     * @param name the name of the material in the above format
+     * @return the material associated with the name
+     */
+    Material getMaterial(String name);
```

Comment:
Same comment as above, why not have this take a RL?

Reply:
The main purpose of this method is to be used when reading and writing a material to NBT, or additionally for a material lookup from CraftTweaker or GroovyScript. Introducing a ResourceLocation into this method leads to unnecessary object creation during each read/write and tweaker lookup.